lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSIdentityStorageImpl.java

```diff
@@ -32,35 +69,761 @@
  * Oct 5, 2015  
  */
 public class RDBMSIdentityStorageImpl extends IdentityStorageImpl {
+
+  private ActivityDAO activityDAO = null;
+  private IdentityDAO identityDAO = null;
+  private ProfileDAO profileDAO = null;
+  private SpaceStorage spaceStorage = null;
+
+  private final DataInitializer dataInitializer;
+
+  private ProfileSearchConnector profileSearchConnector;
+
+  public RDBMSIdentityStorageImpl(DataInitializer dataInitializer) {
+    System.out.println("Initial here");
+    this.dataInitializer = dataInitializer;
+  }
+
+  private ActivityDAO getActivityDAO() {
+    if (activityDAO == null) {
+      activityDAO = CommonsUtils.getService(ActivityDAO.class);
```

Comment:
Should be injected in the constructor instead (easier to read and more easily testable). This remark is true for all services.